File: tug-o-war/javascript/1/tug-o-war.js
function tug_o_war(teams) {
  var team1 = 'Team 1 wins!';
  var team2 = 'Team 2 wins!';
  var tie = "It's a tie!";
  var sum1 = sum(teams[0]);
  var sum2 = sum(teams[1]);
  if (sum1 > sum2) return team1;
  if (sum1 < sum2) return team2;
  var anchor1 = teams[0][0];
  var anchor2 = teams[1][4];
  if (anchor1 > anchor2) return team1;
  if (anchor1 < anchor2) return team2;
  return tie;
}

function sum(team) {
  return team[0] + team[1] + team[2] + team[3] + team[4];
}

File: tug-o-war/java/1/TugOWar.java
public class TugOWar {
  public static final String TEAM1 = "Team 1 wins!";
  public static final String TEAM2 = "Team 2 wins!";
  public static final String TIE = "It's a tie!";
  
  public static String tugOWar(final int[][] teams) {
    final int[] team1 = teams[0];
    final int[] team2 = teams[1];
    final int sum1 = sum(team1);
    final int sum2 = sum(team2);
    if (sum1 > sum2) return TEAM1;
    if (sum1 < sum2) return TEAM2;
    final int anchor1 = team1[0];
    final int anchor2 = team2[team2.length - 1];
    if (anchor1 > anchor2) return TEAM1;
    if (anchor1 < anchor2) return TEAM2;
    return TIE;
  }

  private static int sum(final int[] team) {
    return java.util.Arrays.stream(team).sum();
  }
}